project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -1318,37 +1377,25 @@ record ProjectCreationResponse(String projectId) {
   record ProjectUpdateRequest(String projectId, ProjectUpdateRequestBody requestBody,
                               String requestId) implements CacheableRequest {
 
-    public ProjectUpdateRequest {
-      if (projectId == null) {
-        throw new IllegalArgumentException("Project ID cannot be null");
-      }
-      if (projectId.isBlank()) {
-        throw new IllegalArgumentException("Project ID cannot be blank");
-      }
-      if (requestId == null || requestId.isBlank()) {
-        requestId = UUID.randomUUID().toString();
-      }
-    }
-
     public ProjectUpdateRequest(String projectId, ProjectUpdateRequestBody requestBody) {
       this(projectId, requestBody, UUID.randomUUID().toString());
     }
 
+    public ProjectUpdateRequest {
+      requireNonNull(projectId);
+      requestId =
+          isNull(requestId) || requestId.isBlank() ? UUID.randomUUID().toString() : requestId;
```

Comment:
i think this is a open door for nasty side effects. The client either provides the ID or must explicitly call the constructor method for creating an ID automatically.